zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/CommandBase.java

```diff
@@ -28,22 +28,31 @@ public abstract class CommandBase implements Command {
     private final Set<String> names;
     private final String doc;
     private final boolean serverRequired;
+    private final AuthRequest authRequest;
 
     /**
      * @param names The possible names of this command, with the primary name first.
      */
     protected CommandBase(List<String> names) {
-        this(names, true, null);
+        this(names, true);
     }
     protected CommandBase(List<String> names, boolean serverRequired) {
         this(names, serverRequired, null);
     }
 
-    protected CommandBase(List<String> names, boolean serverRequired, String doc) {
+    protected CommandBase(List<String> names, boolean serverRequired, AuthRequest authRequest) {
+        this(names, serverRequired, null, authRequest);
+    }
+
+    protected CommandBase(List<String> names, boolean serverRequired, String doc, AuthRequest authRequest) {
```

Comment:
nit: It seems `doc` is always null and can be safely removed.

Reply:
Doc is always null and not be used.  Removed.